lxd/instance/drivers/driver_qemu.go

```diff
@@ -2233,6 +2241,52 @@ echo "To start it now, unmount this filesystem and run: systemctl start lxd-agen
 	return nil
 }
 
+func (d *qemu) writeInterfaceConfig(configDrivePath string) error {
+	nics := make(map[string]instance.NICConfig)
+	for deviceName, dev := range d.expandedDevices {
+		if dev["type"] == "nic" {
+			hwaddr, ok := d.localConfig[fmt.Sprintf("volatile.%s.hwaddr", deviceName)]
+			if !ok {
+				continue
+			}
+
+			var networkName string
+			if dev["network"] != "" {
```

Comment:
Have a look at the qemu driver's `addNetDevConfig` which is generating the QMP commands to add the NIC to the VM, based on the properties delivered from the NIC driver itself. This seems to have almost what you need (devName and hwaddr), but is lacking mtu. I think we should prove the concept with devName first, by generating/storing the nics json file inside that function (possibly it may be easier to generate a json file per-nic inside a nics directory in the share). 

Then once we've proved the concept with NIC name we can see about getting the NIC driver to pass the mtu, or perhaps simply read the NIC's host side interface to get it.